core/profile.c

```diff
@@ -808,12 +809,12 @@ static void calculate_sac(struct dive *dive, struct divecomputer *dc, struct plo
 	const struct event *ev = NULL;
 	bool *gases, *gases_scratch;
 
-	gases = malloc(MAX_CYLINDERS * sizeof(*gases));
-	memset(gases, 0, MAX_CYLINDERS * sizeof(*gases));
+	gases = malloc(pi->nr_cylinders * sizeof(*gases));
+	memset(gases, 0, pi->nr_cylinders * sizeof(*gases));
```

Comment:
I'm not a huge fan of this commit, simply because it's so *huge*. But the only real comment I have about it is that I wish you didn't do that blind malloc() conversion.

This should probably have been a calloc() even originally, but at least the size used to be a constant. Now it isn't, and it most definitely should be a calloc().

So those two lines should be

     gases = calloc(pi->nr_cylinders, sizeof(*gases));

instead, which initializes things to zero, and will probably also avoid the warning about possible multiplication overflow.

Reply:
Thanks for review - I will change that to calloc. I'm not familiar with C-style memory management.